app/service_objects/connect_with_oauth_provider.rb

```diff
@@ -18,6 +22,14 @@ def connect
 
   private
 
+  def whitelisted
+    return true if whitelist.empty?
+
+    whitelist.select do |list_domain|
```

Comment:
Can you leave a comment over this block explaining what it does? It's not immediately obvious to me (though I think I understand it), so for future maintenance I feel like a comment here will be valuable.